trinity/protocol/common/requests.py

```diff
@@ -36,12 +36,12 @@ def generate_block_numbers(self, block_number: BlockNumber=None) -> Tuple[BlockN
 
         max_headers = min(self.max_size, self.max_headers)
 
-        return sequence_builder(
+        return cast(Tuple[BlockNumber, ...], sequence_builder(
```

Comment:
Could we update `sequence_builder` to use a generic type so that if `block_number` is a `BlockNumber` it returns `Tuple[BlockNumber, ...]`

Reply:
I don't think that Python generics allow us to define the right *bounds* for that. We basically need to constrain `T` to types that allow `int` based operations such as `+` and `range`. My gut feeling says, it's not possible with `mypy` / `Python` as of today but I'd be impressed to see if someone can come up with something :)

Reply:
I think that this works.

```python
T_num = TypeVar('T_num', bound=int)
```

Reply:
Some brief tests suggest that works, though you'll still need to add a `cast` within the final comprehension of the `sequence_builder` function to force the return values to be of that type.

Reply:
True! That works :+1: